### ISSUE TITLE
Refactor CLI to use cobra

### ISSUE THREAD

**Issue by twhiston** [ansible-semaphore/semaphore#513]
I think it might be beneficial to refactor the cli part of semaphore using Cobra. https://github.com/spf13/cobra
This would allow us to separate the different commands out, making the architecture cleaner, and allowing us to remove some code from the utils package (which i would eventually like to remove, see https://twitter.com/bradfitz/status/975004801288335360 ;) ). Moving the command handling logic out to another package would reduce the number of global variable that we have (see util/config.go) and would simplify the ConfigInit and main functions.
When considering the switch to Cobra we should also consider it's default config handling library Viper (https://github.com/spf13/viper). This would allow us to have a config file in yaml as well as json (and numerous other formats), it would also give us environmental variables for config 'for free' as its simply a matter of activating it in viper.
I think these two changes would allow us to clean up the cli architecture considerably.

**Comment by strangeman** [comment]
I worked with `viper` before - this is the really useful library.

**Comment by twhiston** [comment]
yeah, i think it could help a lot. The 'downside' is that it uses the pflags package under the hood so `-setup` will stop working, it would need to be `--setup`.
Actually in that example it would be it's own command and not an option, but I hope you get my point. This is enough of a breaking change that semantic versioning would require we move to 3.x

**Comment by fiftin** [comment]
Done in v2.7.6
